### PR TITLE
tear down

### PR DESCRIPTION
**Pull request by ericpaulsen**
hey team - this is a draft PR for the tear down page on `docs`. happy to hear any & all feedback. 3 points I would like some clarification on:

1. where should this live? I have it under `setup` but might also be relevant under `admin`... just some thoughts.

2. lines 69-71 - I wasn't able to get the code block in a single line due to the linter's 80 character limit. any ideas as to how to make it so the command is only a single line?

3. line 74 - I was thinking we could ask for feedback at the end of this doc, as the customer who is tearing down might be doing so for a specific reason (feature is lacking, bugs, etc.). where would be the best place to send our customers to provide feedback?

thanks.

### CODE REVIEW

File: setup/teardown.md
Comment (IAmHughes) [inline setup/teardown.md:72]:
```suggestion
```bash
gcloud beta container --project $PROJECT_ID clusters delete \
$CLUSTER_NAME --zone <zone>
```
```

You just add a `\` to make multi-line commands on command line.

File: setup/teardown.md
Comment (IAmHughes) [inline setup/teardown.md:74]: I don't think this should be here. This teardown is something that could be literally just reprovisioning infrastructure to their trial ending. I don't think this feedback here (versus other existing mechanisms) makes as much sense, but open to others' thoughts.

File: setup/teardown.md
Comment (IAmHughes) [inline setup/teardown.md:10]:
```suggestion
include teardown steps for any additional resources that may have been created
outside of these steps
```

Just a suggestion, but it's not necessarily cluster resources. Things like IAM settings and other network security items.

**Comment by khorne3** [comment]
@ericpaulsen I'll take a look and get everything cleaned up too. I wouldn't mind having customers leave feedback in this repo (maybe they could open a new issue?)

File: setup/teardown.md
Comment (IAmHughes) [inline setup/teardown.md:47]:
```suggestion
RESOURCE_GROUP="<MY_RESOURCE_GROUP>"
```

This makes more sense to me, though we may want to change the wording above instead or in addition to. This suggestions sets the variable and is explicit regardless of if you've lost that session in your terminal at some point.

The way it's worded with `echo`, would just show if a value is set. Maybe including the other part of the doc you got this from that states to set the value if it's not set?

File: setup/teardown.md
Comment (IAmHughes) [inline setup/teardown.md:65]: Similar to the Azure comment, may be worth also providing the commands to set the values or changing this to "set the values" instead of "check the values"

File: setup/teardown.md
Comment (IAmHughes) [inline setup/teardown.md:32]: This needs an additional step:

You need to ensure `kubectl` is pointed to the same context still. Add steps to check and set the context to the right cluster before you run the delete command, or at least info to make sure people aren't blinding deleting services.

Maybe something like this would work:

```suggestion
3. Delete any services that have an `EXTERNAL-IP` value
in your namespace by running the following command:

```bash
kubectl delete svc <service-name>
```
```

**Review by IAmHughes — CHANGES_REQUESTED** [review]
@ericpaulsen @khorne3 I've added quite a few comments with suggestions. If you're okay with the suggestion you can just commit it, but otherwise open to feedback on those too.

File: setup/teardown.md
Comment (ericpaulsen) [reply setup/teardown.md:47]: Yes - I left out the steps to set the value if it's not set, to try to keep things brief. However, I think over communicating a step or two will help make things clearer and facilitate more self-service.

File: setup/teardown.md
Comment (ericpaulsen) [reply setup/teardown.md:65]: Noted - will add this in.

File: setup/teardown.md
Comment (ericpaulsen) [reply setup/teardown.md:10]: Makes sense - I'll include.

File: setup/teardown.md
Comment (bpmct) [inline setup/teardown.md:11]: That being said, if there are other resources on a cluster, they will likely be deleted as well. 

Coder's recommended install path involved Coder being installed on to its own separate namespace. Perhaps the teardown docs could discuss how to delete all of the _Coder_ resources, and then separately how to destroy the cluster too?

**Comment by vapurrmaid** [comment]
@ericpaulsen run the below to update your contributions for this :sparkles: 
```
yarn all-contributors add ericpaulsen content
```

(ensure you have first run `yarn` to install the node_modules).

**Comment by bpmct** [comment]
> @ericpaulsen run the below to update your contributions for this ✨
> 
> ```
> yarn all-contributors add ericpaulsen content
> ```
> 
> (ensure you have first run `yarn` to install the node_modules).

@vapurrmaid what is the best way to do this automatically?

**Comment by vapurrmaid** [comment]
> > @ericpaulsen run the below to update your contributions for this ✨
> > ```
> > yarn all-contributors add ericpaulsen content
> > ```
> > 
> > 
> > (ensure you have first run `yarn` to install the node_modules).
> 
> @vapurrmaid what is the best way to do this automatically?

Someone with admin rights has to activate the bot in this repository. Then just use the bot via a comment on the PR to add the contrib

**Comment by ericpaulsen** [comment]
do I just run `yarn all-contributors add ericpaulsen content` on the master branch?

**Comment by vapurrmaid** [comment]
> do I just run `yarn all-contributors add ericpaulsen content` on the master branch?

On this branch, so the update lands with the contrib

**Comment by ericpaulsen** [comment]
received the following: 
`yarn run v1.22.10
warning package.json: No license field
error Command "all-contributors" not found.`

**Comment by vapurrmaid** [comment]
> received the following:
> `yarn run v1.22.10 warning package.json: No license field error Command "all-contributors" not found.`

Might mean you haven't run `yarn` to install?

I just performed the actions as a check:

```
greyb ~/Code/docs (vapurrmaid/chore-contributors) $ git checkout ericpaulsen/teardown 
Branch 'ericpaulsen/teardown' set up to track remote branch 'ericpaulsen/teardown' from 'origin'.
Switched to a new branch 'ericpaulsen/teardown'
greyb ~/Code/docs (ericpaulsen/teardown) $ yarn
yarn install v1.22.5
warning package.json: No license field
warning No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...

Done in 0.35s.
greyb ~/Code/docs (ericpaulsen/teardown) $ yarn all-contributors add ericpaulsen content
yarn run v1.22.5
warning package.json: No license field
$ /home/greyb/Code/docs/node_modules/.bin/all-contributors add ericpaulsen content
Done in 0.28s.
```

Don't sweat it if it doesn't work - we'll have the all contributors bot up soon (thank you @bpmct). We'll make sure you get recorded!

File: setup/teardown.md
Comment (ericpaulsen) [reply setup/teardown.md:11]: Good point - 

> > received the following:
> > `yarn run v1.22.10 warning package.json: No license field error Command "all-contributors" not found.`
> 
> Might mean you haven't run `yarn` to install?
> 
> I just performed the actions as a check:
> 
> ```
> greyb ~/Code/docs (vapurrmaid/chore-contributors) $ git checkout ericpaulsen/teardown 
> Branch 'ericpaulsen/teardown' set up to track remote branch 'ericpaulsen/teardown' from 'origin'.
> Switched to a new branch 'ericpaulsen/teardown'
> greyb ~/Code/docs (ericpaulsen/teardown) $ yarn
> yarn install v1.22.5
> warning package.json: No license field
> warning No license field
> [1/4] Resolving packages...
> [2/4] Fetching packages...
> [3/4] Linking dependencies...
> [4/4] Building fresh packages...
> 
> Done in 0.35s.
> greyb ~/Code/docs (ericpaulsen/teardown) $ yarn all-contributors add ericpaulsen content
> yarn run v1.22.5
> warning package.json: No license field
> $ /home/greyb/Code/docs/node_modules/.bin/all-contributors add ericpaulsen content
> Done in 0.28s.
> ```
> 
> Don't sweat it if it doesn't work - we'll have the all contributors bot up soon (thank you @bpmct). We'll make sure you get recorded!

thank you grey! I ran `git pull` to make sure I had the latest updates, for some reason I'm still receiving the error `Command "all-contributors" not found`.

**Comment by ericpaulsen** [comment]
> > received the following:
> > `yarn run v1.22.10 warning package.json: No license field error Command "all-contributors" not found.`
> 
> Might mean you haven't run `yarn` to install?
> 
> I just performed the actions as a check:
> 
> ```
> greyb ~/Code/docs (vapurrmaid/chore-contributors) $ git checkout ericpaulsen/teardown 
> Branch 'ericpaulsen/teardown' set up to track remote branch 'ericpaulsen/teardown' from 'origin'.
> Switched to a new branch 'ericpaulsen/teardown'
> greyb ~/Code/docs (ericpaulsen/teardown) $ yarn
> yarn install v1.22.5
> warning package.json: No license field
> warning No license field
> [1/4] Resolving packages...
> [2/4] Fetching packages...
> [3/4] Linking dependencies...
> [4/4] Building fresh packages...
> 
> Done in 0.35s.
> greyb ~/Code/docs (ericpaulsen/teardown) $ yarn all-contributors add ericpaulsen content
> yarn run v1.22.5
> warning package.json: No license field
> $ /home/greyb/Code/docs/node_modules/.bin/all-contributors add ericpaulsen content
> Done in 0.28s.
> ```
> 
> Don't sweat it if it doesn't work - we'll have the all contributors bot up soon (thank you @bpmct). We'll make sure you get recorded!

thank you grey! I ran `git pull` to make sure I had the latest updates, for some reason I'm still receiving the error `Command "all-contributors" not found`.

**Comment by khorne3** [comment]
I installed the AllContributors bot

**Comment by vapurrmaid** [comment]
@all-contributors please add @ericpaulsen  for content

File: setup/teardown.md
Comment (ericpaulsen) [reply setup/teardown.md]: Understood - so this would involve adding steps to delete the namespace before the cluster delete commands are run. @khorne3 what are your thoughts?

Comment (khorne3) [reply setup/teardown.md]: That's not a bad idea -- I think it betters match our happy path install (but in reverse obviously)

Comment (ericpaulsen) [reply setup/teardown.md]: so I've just spoken with @sreya - we actually do not have a way to delete all coder-related resources in the cluster, while keeping other unrelated resources intact (but it is possible via a script). if we were to just delete the namespace, it wouldn't remote some cluster-scoped resources associated with coder, which would actually cause a reinstall to fail.

so for now, an alternative to simply deleting the whole cluster is to run `helm uninstall coder`, which will delete the coder services but keep the user environments (and their related disk space) intact.

what are the team's thoughts? @bpmct @khorne3

Comment (bpmct) [reply setup/teardown.md]: Thanks for the context. With that being said, I think that we can keep these instructions and then potentially add the script if/when is is ready.

I do think it's important to mention that deleting this teardown guide removes the whole cluster, which may potentially contain non-coder applications if the user installed extra stuff.

Comment (ericpaulsen) [reply setup/teardown.md]: Agreed - we do have language in this doc that says `This guide shows you how to tear down Coder and the cluster it is deployed on.` So that should be pretty clear to the user. 

As for the `helm uninstall` route, we can add a note saying that it will _only_ delete Coder `services` within the cluster, and all user environments will persist.

Comment (khorne3) [reply setup/teardown.md]: Can we adjust the callout to the following and if anyone asks, note that anything more will require some type of script? I'm afraid of spending a lot of time on something that will need modifications more often than not

> These instructions help you remove infrastructure created when following our
[Kubernetes setup tutorials](../setup/kubernetes/index.md). They do not include
teardown steps for any additional resources that you create. If you need to keep
your cluster, you can run `helm uninstall coder`, which deletes all Coder
services but retains environments and their associated disk space.

Comment (ericpaulsen) [reply setup/teardown.md]: That works well. I'm good with the changes above.

**Review by khorne3 — APPROVED** [review]

**Comment by ericpaulsen** [comment]
Looks like @IAmHughes's requested changes are blocking the merge. I believe we resolved his requested changes? Let me know if I'm missing something here.

**Comment by khorne3** [comment]
@IAmHughes can you review + approve if all is well, please?

**Pull request merged**
